tests/integration/test_basic_examples.py

```diff
@@ -0,0 +1,46 @@
+import aiohttp
+import pytest
+
+from .actions import encoder, screen, temperature, network
+from simulator import MachineType, Thermistor, Printer
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture
+def wui_base_url(printer):
+    return f'http://localhost:{network.proxy_http_port_get(printer)}'
+
+
+@pytest.fixture
+async def wui_client(wui_base_url):
+    async with aiohttp.ClientSession(base_url=wui_base_url) as session:
+        yield session
+
+
+async def test_max_temp_error_on_bed(printer: Printer):
+    await screen.wait_for_text(printer, 'HOME')
+    await temperature.set(printer, Thermistor.BED, 230)
+    await screen.wait_for_text(printer, 'MAXTEMP ERROR')
+
+
+async def test_web_interface_is_accessible(printer: Printer,
+                                           wui_client: aiohttp.ClientSession):
+    await screen.wait_for_text(printer, 'HOME')
+    response = await wui_client.get('/')
+    assert response.ok
+
+
+async def test_filebrowser_shows_files(printer_factory, printer_flash_dir):
+    # prepare files on the flash drive
+    filenames = ['first file.gcode', 'second file.gcode']
+    for filename in filenames:
+        (printer_flash_dir / filename).write(b'')
```

Comment:
Huh? This actually works in python? :-O

Reply:
Python 3.4 – [object oriented filesystem](https://docs.python.org/3/library/pathlib.html) ☺️